zkevm-circuits/src/pi_circuit.rs

```diff
@@ -873,11 +969,92 @@ impl<F: Field> PiCircuitConfig<F> {
         self.q_keccak.enable(region, data_hash_row)?;
         offset += 1;
 
+        //////////////////////////////////////////////
+        ///////// assign l1 block hashes bytes ///////
+        //////////////////////////////////////////////
+        let q_l1_block_hashes_start_row = offset;
+        let q_l1_block_hashes_end_row = q_l1_block_hashes_start_row + KECCAK_DIGEST_SIZE * public_data.max_l1_block_hashes;
+        let num_l1_block_hashes = public_data.get_num_all_l1_block_hashes();
+        let cum_l1_block_hashes = public_data.cum_l1_block_hashes.clone();
+        let dummy_l1_block_hash = get_dummy_l1_block_hash();
+        let mut l1_block_hashes_bytes_rlc = None;
+        let mut l1_block_hashes_bytes_length = None;
+        self.assign_rlc_start(region, &mut offset, &mut rpi_rlc_acc, &mut rpi_length_acc)?;
+
+        for (i, l1_block_hash) in cum_l1_block_hashes
+            .into_iter()
+            .chain(
+                (0..public_data.max_l1_block_hashes - num_l1_block_hashes)
+                    .into_iter()
+                    .map(|_| dummy_l1_block_hash),
+            )
+            .enumerate()
```

Comment:
```suggestion
        for (i, l1_block_hash) in cum_l1_block_hashes
            .into_iter()
            .chain(std::iter::repeat(dummy_l1_block_hash))
            .take(public_data.max_l1_block_hashes)
            .enumerate()
```
is a better way to write the same logic